lib/scenic/index.rb

```diff
@@ -22,15 +22,45 @@ class Index
     #   "CREATE INDEX index_users_on_email ON users USING btree (email)"
     attr_reader :definition
 
+    # The schema under which the index is defined
+    # @return [String]
+    attr_reader :schema_name
+
     # Returns a new instance of Index
     #
     # @param object_name [String] The name of the object that has the index
     # @param index_name [String] The name of the index
     # @param definition [String] The SQL statements that defined the index
-    def initialize(object_name:, index_name:, definition:)
+    def initialize(object_name:, index_name:, definition:, schema_name:)
       @object_name = object_name
       @index_name = index_name
       @definition = definition
+      @schema_name = schema_name
+    end
+
+    # Return a new instance of Index with the definition changed to create
+    # the index against a different object name.
+    #
+    # @param object_name [String] The name of the object that has the index
+    def with_other_object_name(object_name)
+      type = if @definition.start_with? "CREATE UNIQUE"
+        "CREATE UNIQUE INDEX"
+      else
+        "CREATE INDEX"
+      end
+      old_prefix = "#{type} #{@index_name} ON #{@schema_name}.#{@object_name}"
+      new_prefix = "#{type} #{@index_name} ON #{@schema_name}.#{object_name}"
+      unless @definition.start_with? old_prefix
+        raise "Unhandled index definition: '#{@definition}'"
+      end
+      suffix = @definition.slice((old_prefix.size)..(@definition.size))
+      tweaked_definition = new_prefix + suffix
+      self.class.new(
+        object_name: object_name,
+        index_name: @index_name,
+        schema_name: @schema_name,
+        definition: tweaked_definition
+      )
     end
```

Comment:
This object is not part of the postgres adapter, so we can't do this here.

Reply:
**That's** what I felt like wasn't done yet 🤣